src/ui/RoomSettings.cpp

```diff
@@ -404,6 +442,38 @@ RoomSettings::openEditModal()
     });
 }
 
+void
+RoomSettings::saveHiddenEventsSettings(const bool toggleRoomMember,
+                                       const bool toggleRoomPowerLevels,
+                                       const bool toggleSticker)
+{
+    const auto roomid = roomid_.toStdString();
+    nhlog::ui()->debug("Setting events to hidden in room {}: m.room.member={}, "
+                       "m.room.power_levels={}, m.sticker={}",
+                       roomid,
+                       toggleRoomMember,
+                       toggleRoomPowerLevels,
+                       toggleSticker);
+
+    mtx::events::account_data::nheko_extensions::HiddenEvents hiddenEvents;
+    hiddenEvents.hidden_event_types = {
+      EventType::Reaction, EventType::CallCandidates, EventType::Unsupported};
+    if (toggleRoomMember) {
+        hiddenEvents.hidden_event_types.emplace_back(mtx::events::EventType::RoomMember);
+    }
+    if (toggleRoomPowerLevels) {
+        hiddenEvents.hidden_event_types.emplace_back(mtx::events::EventType::RoomPowerLevels);
+    }
+    if (toggleSticker) {
+        hiddenEvents.hidden_event_types.emplace_back(mtx::events::EventType::Sticker);
+    }
+    http::client()->put_room_account_data(roomid, hiddenEvents, [&roomid](mtx::http::RequestErr e) {
```

Comment:
It might make sense to make this independent of the RoomSettings, so that you can reuse the same code, just with empty or set roomid.